chainer/functions/pooling/average_pooling_2d.py

```diff
@@ -23,6 +24,7 @@ def forward_cpu(self, x):
 
     def forward_gpu(self, x):
         if chainer.should_use_cudnn('>=auto'):
+            self.x, = x
```

Comment:
It's better to use `retain_inputs` instead of saving the input array manually.

Reply:
Thanks, I fixed.